CodeEdit/Features/CEWorkspace/Models/CEWorkspaceFileManager.swift

```diff
@@ -187,33 +187,28 @@ final class CEWorkspaceFileManager {
     /// - Parameter events: An array of events that occurred.
     private func fileSystemEventReceived(events: [DirectoryEventStream.Event]) {
         DispatchQueue.main.async {
-            var eventHandledCount = 0
+            var files: Set<CEWorkspaceFile> = []
             for event in events {
-                var directory = event.directory
-                if directory.last == "/" {
-                    directory.removeLast()
-                }
-                guard let item = self.getFile(directory) else {
+                // Event returns file/folder that was changed, but in tree we need to update it's parent
+                let parent = "/" + event.path.split(separator: "/").dropLast().joined(separator: "/")
```

Comment:
This looks a bit like a overkill especially compared with the old code `directory.removeLast()`

Reply:
This does different thing

`/folder/file` -> `/folder` vs `/folder/` -> `/folder`